### PR TITLE
Implemented-flood-control

### PR DESCRIPTION
**Pull request by vg3095**
Implemented flood controls using Drupal flood service and utilized user_login_finalize for additional flood control stuff.

### CODE REVIEW

**Comment by therealssj** [comment]
https://github.com/d8-contrib-modules/tfa/pull/17
This pull request.
We are not using TFAManager class anymore, I don't think we have any use for these changes here.

File: src/TfaManager.php [hunk 1]
```diff
@@ -225,14 +225,14 @@ public function setContext($account, $context) {
...
+    $user = $account;
+    user_login_finalize($user);
     // Truncate flood for user.
-    //flood_clear_event('tfa_begin');
-    //$identifier = variable_get('user_failed_login_identifier_uid_only', FALSE) ? $account->uid : $account->uid . '-' . ip_address();
-    //flood_clear_event('tfa_user', $identifier);
-    //$edit = array();
-    //user_module_invoke('login', $edit, $user);
+    \Drupal::flood()->clear('tfa_begin');
+    $identifier = \Drupal::config('tfa.settings')->get('user_failed_login_identifier_uid_only', FALSE) ? $account->uid : $account->uid . '-' . Drupal::request()->getClientIp();
```
Comment (therealssj) [inline src/TfaManager.php:234]: I don't think this is a defined configuration.
Does this even need to be a configuration, this feels to be more like user dependent.

**Comment by vg3095** [comment]
I read this issue #12  .That`s why I implemented it.

**Comment by therealssj** [comment]
I do not intend to say it is wrong.
I am just pointing out the fact that we are not using TFAManager any more.
Issue #12 states that flood control doesn't work.
It does not say where the fix has to be done.

**Comment by vg3095** [comment]
So, I think this should become part of Login plugin manager then

**Comment by therealssj** [comment]
Umm I am not sure.
Would be good if we discuss this here https://github.com/d8-contrib-modules/tfa/issues/12
Drop your thoughts on this so others can also contribute :)

**Pull request closed without merging**